feat: add pnpm support

penx opened the pull request:
This is branched @FFdhorkin's work at https://github.com/FFdhorkin/github-action/tree/feature/pnpm-support but currently still has a few issues.


Fixes #145

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/github-action?pullRequest=381) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/github-action?pullRequest=381) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/github-action?pullRequest=381) it.</sub>

lifeiscontent commented:
@jennifer-shehane any idea when the cypress team might take a look at this PR? Really would love to use these improvements.

FFdhorkin commented:
@penx regarding the error you are getting:
>Unable to process command '::add-path::/home/runner/setup-pnpm/.bin' successfully.
>
>The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`.

Perhaps this is due to the older version of PNPM's github action in this PR? The example file uses `pnpm/action-setup@v1.2.0` but the latest is `pnpm/action-setup@v2.0.1` (not to mention pnpm 5.4.3 is superceded by 6.3.x now)

ebeloded commented on examples/basic-pnpm/package.json:
```diff
@@ -0,0 +1,16 @@
+  "license": "ISC",
+  "private": true,
+  "devDependencies": {
+    "cypress": "4.8.0"
```
The current version of Cypress is 7.5

ebeloded commented:
How can use this version of the action?

penx commented:
Hi all, just to let you know I'm taking a break for a couple weeks and will pick this up again after that. If someone wants to pick it up in the meantime they are more than welcome to.

Also check #382 which is what I personally need and is very similar.

lifeiscontent commented:
@bahmutov can you take a look at this PR?

penx commented:
I've:
- updated the following dependencies to the latest versions
  - cypress
  - pnpm/action-setup
  - pnpm
- Rebased off master and rebuilt dist
- Updated the pnpm lock file

but this still fails with

```
Run ./
  with:
    working-directory: examples/basic-pnpm
    build: pnpx cypress info
    record: false
    config-file: cypress.json
/home/runner/setup-pnpm/node_modules/.bin/pnpm install --frozen-lockfile -C
undefined
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command failed with ENOENT: install --frozen-lockfile -C
spawn install --frozen-lockfile -C ENOENT
Error: The process '/home/runner/setup-pnpm/node_modules/.bin/pnpm' failed with exit code 1
```
https://github.com/penx/github-action/runs/3589281020?check_suite_focus=true

penx commented:
This was intentionally in draft as I was picking up someone else's PR to see if it would build on CI with some minor modifications.

It didn't, and I'm not sure what to do next to resolve the issues I'm afraid. I don't personally need pnpm support but will be happy to try out any suggestions - or even better if someone wants to pick it up then feel free.

penx commented:
I think the comment here https://github.com/cypress-io/github-action/issues/145#issuecomment-884808726 may also be the way forward - i.e. if you are using pnpm, manage the dependency install yourself and then pass `install:false` to this action.

penx commented:
Closing in favour of the workaround detailed above
